File: 프로그래머스/1/72410. 신규 아이디 추천/신규 아이디 추천.cpp
#include <string>
#include <iostream>
#include <regex>
#include <vector>

using namespace std;

string solution(string new_id) {
    string answer ="";
    
    for (char &ch : new_id)
        ch = tolower(ch);
    
    answer = regex_replace(new_id, regex("[^a-z0-9\\-_.]"), "");
    
    answer = regex_replace(answer, regex("\\.+"), ".");

    answer = regex_replace(answer, regex("^[.]|[.]$"), "");

    if (answer.empty()) 
        answer = "a";

    if (answer.size() > 15){
        answer = answer.substr(0, 15);
        answer = regex_replace(answer, regex("[.]$"), "");
    }
    
    while (answer.size() < 3) 
        answer += answer.back();
    
    return answer;
}

File: 프로그래머스/1/72410. 신규 아이디 추천/신규 아이디 추천.java
class Solution {
    public String solution(String new_id) {
        String answer = "";
        new_id = new_id.toLowerCase();
        new_id = new_id.replaceAll("[^-_.a-z0-9]","");
        new_id = new_id.replaceAll("\\.{2,}", ".");
        new_id = new_id.replaceAll("^\\.|\\.&", "");
        new_id = new_id.isEmpty() ? "a" : new_id;
        if (new_id.length() >= 16) {
		new_id = new_id.substring(0, 15);
}       
        new_id = new_id.replaceAll("[.]$", "");
        if (new_id.length() <= 2)
		while (new_id.length() < 3)
				new_id += new_id.charAt(new_id.length()-1);
        answer = new_id;
        return answer;
    }
}